chartfx-dataset/src/test/java/de/gsi/dataset/spi/DataSetBuilderTests.java

```diff
@@ -96,6 +98,105 @@ public void testYErrorDataSetNoCopy() {
         assertArrayEquals(new double[] { 0.1, 0.2, 0.1 }, errorDataSet.getErrorsNegative(DIM_Y));
         assertEquals(ErrorType.NO_ERROR, errorDataSet.getErrorType(DIM_X));
         assertEquals(ErrorType.ASYMMETRIC, errorDataSet.getErrorType(DIM_Y));
+        // noCopy
+        final DataSet dataset3 = new DataSetBuilder() //
+                                         .setName("testdataset") //
+                                         .setXValuesNoCopy(new double[] { 1, 2, 3 }) //
+                                         .setYValuesNoCopy(new double[] { 1.337, 23.42, 0.0 }) //
+                                         .setYPosErrorNoCopy(new double[] { 0.1, 0.2, 0.1 }) //
+                                         .setYNegErrorNoCopy(new double[] { 0.1, 0.2, 0.1 }) //
+                                         .build();
+        assertEquals("testdataset", dataset3.getName());
+        assertEquals(3, dataset3.getDataCount());
+        assertEquals(3, dataset3.getDataCount(DIM_X));
+        assertEquals(3, dataset3.getDataCount(DIM_Y));
+        assertArrayEquals(new double[] { 1, 2, 3 }, dataset3.getValues(DIM_X));
+        assertArrayEquals(new double[] { 1.337, 23.42, 0.0 }, dataset3.getValues(DIM_Y));
+        assertEquals(2, dataset3.getDimension());
+        assertThrows(IndexOutOfBoundsException.class, () -> dataset3.get(DIM_X, 3));
+        assertTrue(dataset instanceof DataSetError);
+        final DataSetError errordataset3 = (DataSetError) dataset;
+        assertArrayEquals(new double[] { 0, 0, 0 }, errordataset3.getErrorsPositive(DIM_X));
+        assertArrayEquals(new double[] { 0, 0, 0 }, errordataset3.getErrorsNegative(DIM_X));
+        assertArrayEquals(new double[] { 0.1, 0.2, 0.1 }, errordataset3.getErrorsPositive(DIM_Y));
+        assertArrayEquals(new double[] { 0.1, 0.2, 0.1 }, errordataset3.getErrorsNegative(DIM_Y));
+        assertEquals(ErrorType.NO_ERROR, errordataset3.getErrorType(DIM_X));
+        assertEquals(ErrorType.ASYMMETRIC, errordataset3.getErrorType(DIM_Y));
+        // unsupported X errors
+        assertThrows(UnsupportedOperationException.class, () -> {
+            new DataSetBuilder().setXPosError(new double[] { 0.0, 3 }).setXNegError(new double[] { 1, 2 }).setDimension(2).build();
+        });
+        assertThrows(UnsupportedOperationException.class, () -> {
+            new DataSetBuilder().setXPosErrorNoCopy(new double[] { 0.0, 3 }).setXNegErrorNoCopy(new double[] { 1, 2 }).setDimension(2).build();
+        });
+
+        // 3D
+        final DataSet dataset2 = new DataSetBuilder() //
+                                         .setName("testdataset") //
+                                         .setXValuesNoCopy(new double[] { 1, 2, 3 }) //
+                                         .setYValuesNoCopy(new double[] { 1.337, 23.42 }) //
+                                         .setValues(DataSet.DIM_Z, new double[][] { { 1, 2, 3 }, { 4, 5, 6 } }) //
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'DataSet.DIM_Z' due to existing static import 'de.gsi.dataset.DataSet.DIM_Z'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5481451)